R/db-interface.R

```diff
@@ -187,15 +187,14 @@ copy_dm_to <- function(dest, dm, ...,
   # constraints
 
   # use 0-rows dm object from now on
-  dm ## we still need it to copy actual data, we will collect tables one by one to reduce peak memory usage
   ptype_dm <- collect(dm_ptype(dm))
 
   # Shortcut necessary to avoid copying into .GlobalEnv
   if (!is_db(dest)) {
     return(dm)
   }
 
-  queries <- build_copy_queries(dest_con, ptype_dm, set_key_constraints, temporary, table_names_out)
+  queries <- build_copy_queries(dest_con, dm, set_key_constraints, temporary, table_names_out)
```

Comment:
In the end, this should be `dm_sql()` , and the code below should be simplified. Everything needed to run the queries should be contained (including the ticker messages) in the returned object.